### DISCUSSION TITLE
Setting maximum and minimum value in LineChart Y-Axis

### DISCUSSION THREAD

**Discussion by AlexKlueh** [Megabit/Blazorise#1925] [Q&A] [answered]
Hi there,

I want to set minimum and maximum y-values on a linechart, independent of the given data.

How can this be accomplished?

Best regards,

Alexander

**Comment by stsrki** [comment, marked as answer]
Hi, sorry for not answering your question before.

I have looked at some examples on [stackoverflow](https://stackoverflow.com/questions/28990708/how-to-set-max-and-min-value-for-y-axis). You can see in the example below that I only converted JSON to C# `object` and apply it to OptionsObject. With this you can create any object structure and it will be properly serialized.

```cshtml
<LineChart TItem="double" OptionsObject="@chartOptions" />
@code{
    object chartOptions = new
    {
        Scales = new
        {
            YAxes = new[] {
                new {
                    Ticks = new
                    {
                        BeginAtZero = true,
                        Min = 0,
                        Max = 500,
                    }
                }
            }
        }
    };
}
```

**Reply by AlexKlueh** [reply]
Thanks for the update. This works fine with fixed Min and Max values defined at designtime.
Is it possible to change these values dynamically, e.g. when feeding new data into the linechart, then set new Min Max values?

**Reply by stsrki** [reply]
It should work with calling the `chart.SetOptions(null);`. Notice I defined `null `for options. This is fine because you're using `OptionsObject` anyways, and they will be used internally.

**Reply by AlexKlueh** [reply]
When I call chart.SetOptions(null) after feeding new data via chart.AddLabelsDatasetsAndUpdate(..)
and calling chart.Update(), it does not work. The new Max value is in effect only after I call chart.SetOptions(null) later, e.g. from a button click handler which I implemented for test purposes.

Am I doing something wrong?

**Reply by stsrki** [reply]
Probably options must be set before data has changed. Just reorder calls.

**Reply by AlexKlueh** [reply]
I tried this, also setting options before and after. Is it possible that this is related with the Animation duration?

**Reply by edgarssults** [reply]
Thanks, this helped me too. Would be nice if the option data structures were documented or strongly typed.

![image](https://user-images.githubusercontent.com/8101338/139313762-b882c030-7b5d-447d-8646-7d4e56f1b54f.png)

**Comment by AlexKlueh** [comment]
I was finally able to get it running correctly by calling asynchronously:

 Task.Run(() =>
            {
                chart.SetOptions(null);
                chart.Update();
            });

**Reply by stsrki** [reply]
You should have called them always asynchronously :)

```cs
async Task YourEvent()
{
  await chart.SetOptions(null);
  await chart.Update();
}
```

All chart methods should be called like that.
